nifi-nar-bundles/nifi-graph-bundle/nifi-graph-processors/src/main/java/org/apache/nifi/processors/graph/ExecuteGraphQueryRecord.java

```diff
@@ -223,14 +237,19 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
                     }
 
                     dynamicPropertyMap.putAll(input.getAttributes());
+                    if (getLogger().isDebugEnabled()) {
+                        getLogger().info("Parameter map: " + dynamicPropertyMap);
+                    }
                     List<Map<String, Object>> graphResponses = new ArrayList<>(executeQuery(recordScript, dynamicPropertyMap));
 
                     OutputStream graphOutputStream = session.write(graph);
                     String graphOutput = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(graphResponses);
                     graphOutputStream.write(graphOutput.getBytes(StandardCharsets.UTF_8));
                     graphOutputStream.close();
                     session.transfer(graph, GRAPH);
+                    records++;
                 } catch (Exception e) {
+                    getLogger().error("Error processing record", e);
```

Comment:
Is there any type of record identifier that could be included in the log message?  That could be useful when attempting to troubleshoot a problem if more than one record failed during this loop.

Reply:
Not a good one, but I decided to add the record index.